src/cljs/rems/administration/create_catalogue_item.cljs

```diff
@@ -44,42 +53,49 @@
 
-(rf/reg-sub ::selected-resource-id (fn [db _] (get-in db [::form :resource-id])))
-(rf/reg-event-db ::set-selected-resource-id (fn [db [_ resource-id]] (assoc-in db [::form :resource-id] resource-id)))
+(rf/reg-sub ::selected-workflow (fn [db _] (get-in db [::form :workflow])))
+(rf/reg-event-db ::set-selected-workflow (fn [db [_ workflow]] (assoc-in db [::form :workflow] workflow)))
 
-(rf/reg-sub ::selected-form-id (fn [db _] (get-in db [::form :form-id])))
-(rf/reg-event-db ::set-selected-form-id (fn [db [_ form-id]] (assoc-in db [::form :form-id] form-id)))
+(rf/reg-sub ::selected-resource (fn [db _] (get-in db [::form :resource])))
+(rf/reg-event-db ::set-selected-resource (fn [db [_ resource]] (assoc-in db [::form :resource] resource)))
+
+(rf/reg-sub ::selected-form (fn [db _] (get-in db [::form :form])))
+(rf/reg-event-db ::set-selected-form (fn [db [_ form]] (assoc-in db [::form :form] form)))
 
 (defn- valid-localization? [localization]
   (not (str/blank? (:title localization))))
 
-(defn- valid-request? [request languages]
-  (and (number? (:wfid request))
-       (number? (:resid request))
-       (number? (:form request))
-       (= (set languages)
-          (set (keys (:localizations request))))
-       (every? valid-localization? (vals (:localizations request)))))
+(defn- valid-request? [form request languages]
+  (let [selected-organization (:organization request)]
+    (and (= (get-in form [:workflow :organization]) selected-organization)
+         (= (get-in form [:resource :organization]) selected-organization)
+         (= (get-in form [:form :form/organization]) selected-organization)
+         (number? (:wfid request))
+         (number? (:resid request))
+         (number? (:form request))
+         (= (set languages)
+            (set (keys (:localizations request))))
+         (every? valid-localization? (vals (:localizations request))))))
 
 (defn- empty-string-to-nil [str]
   (when-not (str/blank? str)
     str))
 
 (defn build-request [form languages]
-  (let [request {:wfid (:workflow-id form)
-                 :resid (:resource-id form)
-                 :form (:form-id form)
+  (let [request {:wfid (get-in form [:workflow :id])
+                 :resid (get-in form [:resource :id])
+                 :form (get-in form [:form :form/id])
                  :organization (or (:organization form) "")
                  :localizations (into {}
                                       (for [lang languages]
                                         [lang {:title (trim-when-string (get-in form [:title lang]))
                                                :infourl (-> (get-in form [:infourl lang])
                                                             empty-string-to-nil
                                                             trim-when-string)}]))}]
-    (when (valid-request? request languages)
+    (when (valid-request? form request languages)
```

Comment:
how about keeping `valid-request?` as-is and adding a separate check for organizations? not insisting.

Reply:
I might prefer having the organization check as a part of `valid-request?` just by considering the name of the function, since I wouldn't count a request as "valid" if it has incompatible organizations.